### PR TITLE
[Merged by Bors] - feat: port Analysis.SpecialFunctions.Trigonometric.Deriv

### PR DESCRIPTION
**Pull request by kim-em**
- [x] depends on: #4651

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4651
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4651~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Deriv.lean [hunk 1]
```diff
@@ -0,0 +1,1188 @@
...
+  cosh_zero ▸ cosh_le_cosh.2 (by simp only [_root_.abs_zero, _root_.abs_nonneg])
+#align real.one_le_cosh Real.one_le_cosh
+
+@[simp]
+theorem one_lt_cosh : 1 < cosh x ↔ x ≠ 0 :=
+  cosh_zero ▸ cosh_lt_cosh.trans (by simp only [_root_.abs_zero, abs_pos])
+#align real.one_lt_cosh Real.one_lt_cosh
+
+theorem sinh_sub_id_strictMono : StrictMono fun x => sinh x - x := by
+  -- Porting note: `by simp; abel` wass just `by simp` in mathlib3.
```
Comment (dupuisf) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Deriv.lean:750]:
```suggestion
  -- Porting note: `by simp; abel` was just `by simp` in mathlib3.
```

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**